lib/matplotlib/axis.py

```diff
@@ -826,16 +826,23 @@ def set_tick_params(self, which='major', reset=False, **kw):
         For documentation of keyword arguments, see
         :meth:`matplotlib.axes.Axes.tick_params`.
         """
+        blacklist = ['tick1On', 'tick2On', 'label1On', 'label2On', 'gridOn']
         dicts = []
+        blacklist_dicts = []
         if which == 'major' or which == 'both':
             dicts.append(self._major_tick_kw)
         if which == 'minor' or which == 'both':
             dicts.append(self._minor_tick_kw)
         kwtrans = self._translate_tick_kw(kw)
+
         for d in dicts:
             if reset:
                 d.clear()
+            blacklist_dicts.append(
+                    {k: d.pop(k) for k in blacklist if k in d.keys()}
```

Comment:
I'll note that the iteration is only over the keys in the blacklist (total of 7 items), `k in d.keys()` is an O(1) check since keys view objects are set-like and hashable.

The second iteration could actually be rolled into this conditional, to prevent things from being popped then added back with updates, rather than popping indiscriminately and then iterating over the kwtrans keys.